cloudify_cli/commands/apply.py

```diff
@@ -0,0 +1,256 @@
+@cfy.options.ignore_failure
+@cfy.options.install_first
+@cfy.options.preview
+@cfy.options.dont_update_plugins
+@cfy.options.force(help=helptexts.FORCE_UPDATE)
+@cfy.options.tenant_name(required=False,
+                         resource_name_for_help='blueprint and deployment')
+@cfy.options.visibility(mutually_exclusive_required=False)
+@cfy.options.validate
+@cfy.options.include_logs
+@cfy.options.json_output
+@cfy.options.common_options
+@cfy.options.runtime_only_evaluation
+@cfy.options.auto_correct_types
+@cfy.options.reevaluate_active_statuses()
+# From install command
+@cfy.options.skip_plugins_validation
+@cfy.options.parameters
+@cfy.options.allow_custom_parameters
+@cfy.options.blueprint_labels
+@cfy.options.deployment_labels
+@cfy.assert_manager_active()
+@cfy.pass_client()
+@cfy.pass_logger
+@cfy.pass_context
+def apply(ctx,
+          blueprint_path,
+          deployment_id,
+          blueprint_filename,
+          blueprint_id,
+          inputs,
+          reinstall_list,
+          workflow_id,
+          skip_install,
+          skip_uninstall,
+          dont_skip_reinstall,
+          ignore_failure,
+          install_first,
+          preview,
+          dont_update_plugins,
+          force,
+          tenant_name,
+          visibility,
+          validate,
+          include_logs,
+          json_output,
+          runtime_only_evaluation,
+          auto_correct_types,
+          reevaluate_active_statuses,
+          skip_plugins_validation,
+          parameters,
+          allow_custom_parameters,
+          blueprint_labels,
+          deployment_labels,
+          logger,
+          client
+          ):
+    """Apply command uses the `cfy install` or `cfy deployments update`
+    depending on the existence the deployment specified by `DEPLOYMENT_ID`.
+
+    If the deployment exists, the deployment will be updated with the
+    given blueprint.
+    Otherwise the blueprint will installed (the deployment name will be
+    DEPLOYMENT_ID).
+    In both cases the blueprint is being uploaded to the manager.
+
+    `BLUEPRINT_PATH` can be a:
+        - local blueprint yaml file
+        - blueprint archive
+        - url to a blueprint archive
+        - github repo (`organization/blueprint_repo[:tag/branch]`)
+
+    Supported archive types are: zip, tar, tar.gz and tar.bz2
+
+    `DEPLOYMENT_ID` is the deployment's id to install/update.
+
+    Default values:
+
+    If BLUEPRINT_PATH not provided, the default blueprint path
+    'blueprint.yaml' in the current work directory will be used.
+    If DEPLOYMENT_ID is not provided it will be inferred from BLUEPRINT_PATH
+    as follows:
+    - Directory name for local blueprint path or archive with default
+      --blueprint-filename(blueprint.yaml).
+    - <Directory name>.<blueprint_filename> if its an archive and
+      --blueprint-filename is not default.
+    """
+    if not blueprint_path:
+        blueprint_path = blueprint_path or os.path.join(os.getcwd(),
```

Comment:
We might want to handle the case where the blueprint.yaml doesn't exist in the current directory.

Reply:
It's being handled by the `process_blueprint_and_infer_deployment_id` function as for other commands so it fine. Tested this.